tests/utilities/test_reader.py

```diff
@@ -341,3 +341,67 @@ def test_pvdreader_no_time_group():
         assert dataset.time == 0.0
         assert dataset.group == None
         assert dataset.part == i
+
+
+def get_cavity_reader():
+    filename = examples.download_cavity(load=False)
+    return pyvista.get_reader(filename)
+
+
+def test_openfoamreader_arrays_time():
+    reader = get_cavity_reader()
+    assert isinstance(reader, pyvista.OpenFOAMReader)
+
+    assert reader.number_point_arrays == 0
+    assert reader.number_cell_arrays == 2
+
+    assert reader.number_time_points == 6
+    assert reader.time_values == [0.0, 0.5, 1.0, 1.5, 2.0, 2.5]
+
+    # OpenFoamReader currently does not support inspecting active time
+    # Test that read data conforms to time setting
+    mesh = reader.read()
+    assert np.isclose(np.max(mesh[0]['p']), 0.0)
+
+    reader = get_cavity_reader()
+    reader.set_active_time_point(1)
+    mesh = reader.read()
+    assert not np.isclose(np.max(mesh[0]['p']), 0.0)
+
+
+def test_openfoam_cell_to_point_default():
+    reader = get_cavity_reader()
+    mesh = reader.read()
+    assert mesh[0].n_arrays == 4
+
+    reader = get_cavity_reader()
+    reader.cell_to_point_creation = False
+    mesh = reader.read()
+    assert mesh[0].n_arrays == 2
+
+
+def test_openfoam_patch_arrays():
```

Comment:
This test should fail in vtk==9.0.x.  The internal block names for the boundary patches changed from 9.0 to 9.1.  I'm not sure if we should guard against this.

Reply:
`xfail` based on VTK version?

Reply:
Good idea, I need to figure out whether we currently have the ability to inspect vtk minor version , or this needs to be added. Added to TODO.

Reply:
I don't like the idea however that this isn't run on the CI.  We should consider adding a set of tests with supported versions of vtk.

Reply:
As I noted in slack (just forgot to ping/reply, sorry), the check should probably go here: https://github.com/pyvista/pyvista/blob/fd14e16f22db6d485a8bfe4101b39bbcb83a88c6/pyvista/_vtk.py#L11-L18

Reply:
After #1899 , we now can now use `pyvista.vtk_version_info`.  This test now adjusts itself based on the version.

Reply:
I was going to ask if it were cleaner to split this into two version-specific tests, one always to be skipped. But the overlap is large enough that it might not be worth it.

Reply:
The only differences I can find now are in the naming scheme for patch meshes in OpenFOAM datasets.  Otherwise, the logic is exactly the same.  This is why I decided to have a single test, but adjusted for the version.

PS, I haven't tested on 8.2 locally yet, nor do I have the ability.  We might have to adjust again if this fails on 8.2 once the data is available for testing.